client/packages/common/src/utils/regex/RegexUtils.tsx

```diff
@@ -41,7 +41,10 @@ export const RegexUtils = {
     keys?: Array<keyof T>
   ) {
     return (keys ?? (Object.keys(object) as Array<keyof T>)).some(key =>
-      RegexUtils.includes(substring, String(object[key]))
+      RegexUtils.includes(
+        this.escapeChars(substring) ?? '',
```

Comment:
The `?? ''` is just because `escapeChars` returns `string | undefined` right? Would it make sense and define it like:

`escapeChars: (regexString: string): string => {...}`

And then handle the undefined cases on the caller site of `escapeChars`. This avoids this problem here where you have a defined substring and `escapeChars` unnecessarily converts it to undefined.

Reply:
have popped that in 😉